src/com/amazon/ion/IonFloat.java

```diff
@@ -66,13 +66,16 @@ public double doubleValue()
      * recommended to call {@link IonFloat#isNumericValue()} before calling
      * this method.
      *
+     * If you need negative zeros, use {@link #doubleValue()}.
```

Comment:
We could consider adding `public Decimal decimalValue()` to `IonNumber` to support negative zeroes as well.

Reply:
I thought about that, but I figured I'd keep the changes as small as possible for now. `BigDecimal` addresses almost all of the use cases, and has the advantage of being in the JDK (as opposed to our own `Decimal` class).